.golangci.yml

```diff
@@ -0,0 +1,142 @@
+run:
+  timeout: 5m
+output:
+  format: colored-line-number
+linters:
+  enable-all: true
+  disable:
+    - cyclop
+    - exhaustive
+    - exhaustivestruct
+    - execinquery
+    - funlen
+    - gochecknoglobals
+    - gocognit
+    - gocyclo
+    - godox
+    - golint
+    - interfacer
+    - lll
+    - maintidx
+    - ireturn
+    - maligned
+    - nolintlint
+    - prealloc
+    - promlinter
+    - scopelint
+    - tagliatelle
+    - testpackage
+    - paralleltest
+    - varnamelen
+    - wrapcheck
+    - wsl
+
+linters-settings:
+  decorder:
+    dec-order:
+      - type
+      - const
+      - var
+      - func
+    disable-dec-order-check: false
+    disable-init-func-first-check: false
+  dupl:
+    threshold: 100
+  errcheck: # Report about not checking of errors in type assertions: `a := b.(MyStruct)`.
+    check-type-assertions: true
+  exhaustruct:
+    exclude:
+      - 'cobra\.Command$'
+  goconst:   
+    min-len: 2 # Minimal length of string constant.
+    min-occurrences: 2 # Minimum occurrences of constant string count to trigger issue.
+    ignore-tests: true
+    match-constant: false  # Look for existing constants matching the values.
+    numbers: true # Search also for duplicated numbers.
+    min: 2 # Minimum value, only works with goconst.numbers
+    max: 2 # Maximum value, only works with goconst.numbers
+  gofumpt:
+    lang-version: "1.17"
+    extra-rules: true
+  revive:
+    ignore-generated-header: true
+    severity: warning
+    confidence: 0.8
+    rules:
+      - name: atomic
+      - name: context-as-argument
+      - name: context-keys-type
+      - name: defer
+      - name: dot-imports
+      - name: empty-block
+      - name: error-naming
+      - name: error-return
+      - name: error-strings
+      - name: errorf
+      - name: increment-decrement
+      - name: modifies-parameter
+      - name: modifies-value-receiver
+      - name: package-comments
+      - name: range
+      - name: receiver-naming
+      - name: redefines-builtin-id
+      - name: superfluous-else
+      - name: time-naming
+      - name: unreachable-code
+      - name: unused-parameter
+      - name: unused-receiver
+      - name: var-declaration
+      - name: var-naming
+  gomnd:
+    ignored-numbers:
+      - '0o600'
+      - '0o644'
+      - '0o655'
+      - '0o666'
+      - '0o770'
+      - '0o755'
+      - '0o765'
+      - '0o777'
+    ignored-functions:
+      - 'survey.MinLength'
+      - 'survey.MaxLength'
+      - 'args.Error'
+      - 'strconv.ParseInt'
```

Comment:
Do we need to ignore these? Don't think we are using survey anywhere.

Reply:
Yes, the survey one we can remove for now. Will keep ParseInt and Errors?